src/v/storage/mvlog/CMakeLists.txt

```diff
@@ -14,6 +14,7 @@ v_cc_library(
     segment_appender.cc
```

Comment:
> I'm not fully sold on this approach (e.g. vs only managing the segment
appender separately but keeping the underlying readable segment with the
rest of the segments)

are you referring to a design where the segments themselves have no identification of being active or not, and that any "activeness" is only attached to a segment appender object which we control separately?

i could never really decide if i liked the way we do it now where the segment appender is separate. all i really knew is that it bothered me that segments.back() was the active segment.

Reply:
Right, exactly. `segments.back()` also bothered me in `disk_log_impl`, and the explicit separation ended up having some nice properties regarding truncation.

I'm a little wary that the an explicit active segment may add some confusion on the read path (log readers, timequeries, term queries, more? all need to consider an explicit active segment). But it does feel kind of nice along the write path.